RecoTracker/MkFit/interface/MkFitInputWrapper.h

```diff
@@ -0,0 +1,44 @@
+#ifndef RecoTracker_MkFit_MkFitInputWrapper_h
+#define RecoTracker_MkFit_MkFitInputWrapper_h
+
+#include "RecoTracker/MkFit/interface/MkFitIndexLayer.h"
+
+#include <memory>
+#include <vector>
+
+namespace mkfit {
+  class Hit;
+  class Track;
+  class LayerNumberConverter;
+  using HitVec = std::vector<Hit>;
+  using TrackVec = std::vector<Track>;
+}  // namespace mkfit
+
+class MkFitInputWrapper {
+public:
+  MkFitInputWrapper();
+  MkFitInputWrapper(MkFitIndexLayer&& indexLayers,
+                    std::vector<mkfit::HitVec>&& hits,
+                    mkfit::TrackVec&& seeds,
+                    mkfit::LayerNumberConverter&& lnc);
+  ~MkFitInputWrapper();
+
+  MkFitInputWrapper(MkFitInputWrapper const&) = delete;
+  MkFitInputWrapper& operator=(MkFitInputWrapper const&) = delete;
+  MkFitInputWrapper(MkFitInputWrapper&&) = default;
```

Comment:
Thanks for pointing it out, it would be more correct to move the definitions of the move operators to the .cc as well. The reason is that as a precaution I want to keep the mkFit internal headers hidden from ROOT dictionary generation (the headers are rather messy), and only forward-declaring requires the definitions to be out-of-line.

The current setup was working, because ROOT dicrionary generation calls only the default constructor, and the moves are called in files that `#include` the needed mkFit headers.